app/src/main/java/com/github/geohunt/app/ui/components/challenge/MainUserView.kt

```diff
@@ -6,101 +6,107 @@ import androidx.compose.material.*
 import androidx.compose.material.icons.Icons
 import androidx.compose.material.icons.rounded.Notifications
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.draw.clipToBounds
 import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.platform.testTag
 import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
 import com.github.geohunt.app.R
-import com.github.geohunt.app.model.database.api.Challenge
 import com.github.geohunt.app.ui.FetchComponent
 import com.github.geohunt.app.ui.components.LabelledIcon
 import com.github.geohunt.app.ui.components.user.ProfileIcon
 import com.github.geohunt.app.i18n.DateFormatUtils
 import com.github.geohunt.app.i18n.toSuffixedString
 
 @Composable
-internal fun MainUserView(challenge: Challenge) {
+internal fun MainUserView(

```

Comment:
`MainUserView` is a bit of a misleading name. Maybe couldn't we rename it to something else that is more specific to the challenge?